tree/treeplayer/src/TTreePlayer.cxx

```diff
@@ -734,6 +734,7 @@ Int_t TTreePlayer::MakeClass(const char *classname, const char *option)
 
    // Connect output files
    if (!classname) classname = fTree->GetName();
+   const TString vclassname = ROOT::Internal::GetCppName(classname);
```

Comment:
I thought about that, but I opted not to change the filename because:

Imagine that I chose a treename called: "-myTree-". As user, I have no idea what the interpreter does behind the scenes to convert it. If I later call ls -myTree-* I will find the generated files. I don't care much if the internal code has changed the variable names, but I would like to find my files without knowing what conversion was applied. If the filename is changed to _myTree_.C I will not be able to find it (at least not so easily).

Reply:
This is a good point.  Another point of view in that in this example the file "-myTree-.C" now declares a class with a different name (`mImyTreemI`) but I guess this is 'easier' to find out.

Nonetheless we should clarify the purpose of the variable (i.e. `classname` is no longer the class name but something else). i.e. have something both `filestem` and `classname` set to their actual values.

With the current state of the PR we also have the 'issue' that if the user provide the `classname` but it is invalid, we produce invalid code.  So maybe we should use:
```
const TString filestem = classname;
if (classname) {
   auto validname( ROOT::Internal::GetCppName(classname));
   if (validname != classname)
      Error(functionname, "The class name provided ('%s') is no a valid C++ identifier, the code produced will likely fail to compile.", classname);
} else {
   filestem = fTree->GetName();
   classname = ROOT::Internal::GetCppName(fTree->GetName());
}
```